serving/docker/partition/sm_neo_trt_llm_partition.py

```diff
@@ -0,0 +1,163 @@
+            f"Neo environment variable '{exc.args[0]}' expected but not found"
+        )
+
+
+def get_neo_compiler_flags(compiler_options):
+    """
+    Get Neo compiler_flags from the CompilerOptions field
+    """
+    try:
+        # CompilerOptions JSON will always be present, but compiler_flags key is optional
+        compiler_options = json.loads(compiler_options)
+        if not isinstance(compiler_options, dict):
+            raise ValueError("Parsed JSON is not a dictionary")
+        return compiler_options.get("compiler_flags")
+    except Exception as exc:
+        raise InputConfiguration(f"Failed to parse Neo CompilerOptions: {exc}")
+
+
+def verify_neo_compiler_flags(compiler_flags):
+    """
+    Verify that provided compiler_flags is a valid configuration
+    """
+    convert_checkpoint_flags = compiler_flags.get("convert_checkpoint_flags")
+    quantize_flags = compiler_flags.get("quantize_flags")
+    trtllm_build_flags = compiler_flags.get("trtllm_build_flags")
+
+    if trtllm_build_flags is None:
+        raise InputConfiguration("`compiler_flags` were found, but required sub-field `trtllm_build_flags` was not defined."
+                                    " See Neo documentation for more info:"
+                                    " https://docs.aws.amazon.com/sagemaker/latest/dg/neo-troubleshooting.html"
+                                )
+    if convert_checkpoint_flags is None and quantize_flags is None:
+        raise InputConfiguration("`compiler_flags` were found, but neither sub-fields `convert_checkpoint_flags` "
+                                    " or `quantize_flags` were defined, at least one of which must be provided."
+                                    " See Neo documentation for more info:"
+                                    " https://docs.aws.amazon.com/sagemaker/latest/dg/neo-troubleshooting.html"
+                                )
+    if convert_checkpoint_flags is not None and quantize_flags is not None:
+        logging.warning("Both `convert_checkpoint_flags` and `quantize_flags` were provided –"
+                        " `convert_checkpoint_flags` will be used.")
+
+
+def main():
+    """
+    Convert from SageMaker Neo interface to DJL-Serving format for TRT-LLM compilation
+    """
+    logging.basicConfig(stream=sys.stdout, format="%(message)s", level=logging.INFO)
+
+    compilation_error_file = None
+    try:
+        (compiler_options,
+        input_model_directory,
+        compiled_model_directory,
+        compilation_error_file) = get_neo_env_vars()
+
+        # Neo requires that serving.properties is in same dir as model files
+        serving_properties = load_properties(input_model_directory)
+        compiler_flags = get_neo_compiler_flags(compiler_options)
+        kwargs = {}
+        
+        if compiler_flags is not None:
+            # If set, prefer Neo CompilerOptions flags over LMI serving.properties
+            logging.info(f"Using CompilerOptions from Neo")
```

Comment:
`Neo` -> `SageMaker Neo` ?

While this log-line mentions `CompilerOptions from Neo` will be used, It will also be useful to explicitly mention that options from serving.properties will be overridden and those from Neo will be used.